### PR TITLE
VB-158: Feature/buy be

### PR DESCRIPTION
**Pull request by mishabenq**
need some review)

### CODE REVIEW

File: packages/backend/src/repositories/product.ts [hunk 1]
```diff
@@ -125,4 +125,25 @@ export class ProductRepository {
...
+        id,
+        status,
+      },
+    });
+  }
+
+  public async buy(id: string, userId: string, status: ProductStatus) {
+    return await this._dbClient.product.update({
+      where: {
+        id,
```
Comment (Nightlngale) [inline packages/backend/src/repositories/product.ts:141]: question: can we save where { id, status: 'active }

File: packages/backend/src/services/product.ts [hunk 2]
```diff
@@ -83,4 +84,21 @@ export class ProductService {
 
     return productId;
   }
+
+  public async buy({ userId, productId }: AddProductToFavorites) {
```
Comment (Nightlngale) [inline packages/backend/src/services/product.ts:88]: Please add a check whether user is validated

Comment (Nightlngale) [reply packages/backend/src/services/product.ts:88]: you can add his field in auth middleware and check in items/buy router

**Review by Nightlngale — CHANGES_REQUESTED** [review]

File: packages/backend/src/repositories/product.ts (hunk 1 above)
Comment (mishabenq) [reply packages/backend/src/repositories/product.ts:141]: image
as i see, while update we can only search by id. There is no status parameter

File: packages/backend/src/services/product.ts (hunk 2 above)
Comment (mishabenq) [reply packages/backend/src/services/product.ts:88]: image
there is auth middleware in router

File: packages/backend/src/repositories/product.ts (hunk 1 above)
Comment (mishabenq) [reply packages/backend/src/repositories/product.ts:141]: image
image
If this solution it`s okay, i can commit it

**Review by Nightlngale — APPROVED** [review]

**Pull request merged**